experiments/OceanBoxGCM/ocean_gyre.jl

```diff
@@ -88,4 +90,51 @@ function run_ocean_gyre(; imex::Bool = false)
 
 end
 
-run_ocean_gyre(imex = false)
+@testset "$(@__FILE__)" begin
+    boundary_conditions = [
+        (
+            CLIMA.HydrostaticBoussinesq.CoastlineNoSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanFloorNoSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanSurfaceStressForcing(),
+        ),
+        (
+            CLIMA.HydrostaticBoussinesq.CoastlineFreeSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanFloorNoSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanSurfaceStressForcing(),
+        ),
+        (
+            CLIMA.HydrostaticBoussinesq.CoastlineNoSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanFloorFreeSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanSurfaceStressForcing(),
+        ),
+        (
+            CLIMA.HydrostaticBoussinesq.CoastlineFreeSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanFloorFreeSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanSurfaceStressForcing(),
+        ),
+        (
+            CLIMA.HydrostaticBoussinesq.CoastlineNoSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanFloorNoSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanSurfaceNoStressForcing(),
+        ),
+        (
+            CLIMA.HydrostaticBoussinesq.CoastlineFreeSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanFloorNoSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanSurfaceNoStressForcing(),
+        ),
+        (
+            CLIMA.HydrostaticBoussinesq.CoastlineNoSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanFloorFreeSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanSurfaceNoStressForcing(),
+        ),
+        (
+            CLIMA.HydrostaticBoussinesq.CoastlineFreeSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanFloorFreeSlip(),
+            CLIMA.HydrostaticBoussinesq.OceanSurfaceNoStressForcing(),
+        ),
+    ]
+
+    for BC in boundary_conditions
+        run_ocean_gyre(imex = false, BC = BC)
+    end
+end
```

Comment:
@simonbyrne @kpamnany @charleskawczynski , do you guys have any ideas on how to easily switch between running all of these options for CI and only running a single setup interactively? 

Currently, this is the only place we can test that all the different boundary conditions are working... besides `test_ocean_gyre.jl` but I fear that running all of these on Azure would take too long and that the integration time is too short to really test the physics anyways

Reply:
Hm, I would suggest testing BCs with a coarse mesh and for a single timestep (assuming they're time-independent BCs)--if it's BCs that you're testing. At first glance, this looks like an entire simulation, in which case I would suggest breaking each case out into individual tests/experiments and running them in parallel on slurm. That said, we definitely need some sort of unit tests for BCs in general.

Reply:
But the boundary condition can run for a single time step successfully but not produce correct results if run for a longer length of time.  We need some way to catch these errors that are small per timestep but gradually accumulate.

Reply:
If the errors depend on the initial state, maybe we should supply an artificial initial state to check the BCs. Or apply a divergence-free random IC. So long the BCs are not time-dependent, a single step/BC application is all that should be needed to test the BCs.

Reply:
It'd be good to have two tests -- a basic and quick test that provides some confidence that things are working, and a detailed test that assures that things are working. The first will go in merge tests, the second in nightly (which is not up yet).